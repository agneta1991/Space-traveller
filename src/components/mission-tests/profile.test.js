import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../profile';

const mockStore = configureStore([]);

describe('Profile Component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      rockets: {
        rockets: [
          {
            id: '1',
            rocket_name: 'Falcon 1',
            description: 'Description 1',
            imageURL: 'image_url_1.jpg',
            reserved: true,
          },
          {
            id: '2',
            rocket_name: 'Falcon 9',
            description: 'Description 2',
            imageURL: 'image_url_2.jpg',
            reserved: false,
          },
        ],
        status: 'succeeded',
        error: null,
      },
      missions: {
        data: [
          {
            mission_id: '1',
            mission_name: 'Test Mission 1',
            description: 'Test Description 1',
            reserved: true,
          },
          {
            mission_id: '2',
            mission_name: 'Test Mission 2',
            description: 'Test Description 2',
            reserved: false,
          },
        ],
        status: 'succeeded',
        error: null,
      },
    });

    const { container } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
