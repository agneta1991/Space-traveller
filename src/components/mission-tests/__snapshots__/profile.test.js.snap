// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component renders correctly 1`] = `
<div>
  <div
    class="main-profile-div"
  >
    <div
      class="missions-profile-div"
    >
      <div>
        <div>
          <h2>
            My Missions
          </h2>
          <table
            class="missions-profile-table"
          >
            <tbody>
              <tr>
                <td>
                  Test Mission 1
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="rockets-profile-div"
    >
      <div
        class="rocket-profile-div"
      >
        <div>
          <div>
            <h2>
              My Rockets
            </h2>
            <table
              class="rocket-profile-table"
            >
              <tbody>
                <tr>
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
